services/repository/src/main/java/org/sagebionetworks/auth/AuthenticationFilter.java

```diff
@@ -46,126 +43,111 @@
+		if (isTokenEmptyOrNull(sessionToken)) {
 			// Check for a session token as a parameter
 			sessionToken = req.getParameter(AuthorizationConstants.SESSION_TOKEN_PARAM);
 		}
-		
+
 		// Determine the caller's identity
 		Long userId = null;
-		
+		String bearerToken = null;
+
 		// A session token maps to a specific user
-		if (!isSessionTokenEmptyOrNull(sessionToken)) {
+		if (!isTokenEmptyOrNull(sessionToken)) {
 			String failureReason = "Invalid session token";
 			try {
 				userId = authenticationService.revalidate(sessionToken, false);
-			} catch (UnauthenticatedException e) {
-				reject(req, (HttpServletResponse) servletResponse, failureReason);
-				log.warn(failureReason, e);
-				return;
-			} catch (NotFoundException e) {
-				reject(req, (HttpServletResponse) servletResponse, failureReason);
+			} catch (UnauthenticatedException | NotFoundException  e) {
+				HttpAuthUtil.reject((HttpServletResponse) servletResponse, failureReason);
 				log.warn(failureReason, e);
 				return;
 			}
-		
-		// If there is no session token, then check for a HMAC signature
+			bearerToken=oidcTokenHelper.createTotalAccessToken(userId);
+			// If there is no session token, then check for a HMAC signature
 		} else if (isSigned(req)) {
 			String failureReason = "Invalid HMAC signature";
 			String username = req.getHeader(AuthorizationConstants.USER_ID_HEADER);
 			try {
 				userId = userManager.lookupUserByUsernameOrEmail(username).getPrincipalId();
 				String secretKey = authenticationService.getSecretKey(userId);
 				matchHMACSHA1Signature(req, secretKey);
-			} catch (UnauthenticatedException e) {
-				reject(req, (HttpServletResponse) servletResponse, e.getMessage());
-				log.warn(failureReason, e);
-				return;
-			} catch (NotFoundException e) {
-				reject(req, (HttpServletResponse) servletResponse, e.getMessage());
+			} catch (UnauthenticatedException | NotFoundException e) {
+				HttpAuthUtil.reject((HttpServletResponse) servletResponse, e.getMessage());
 				log.warn(failureReason, e);
 				return;
 			}
+			bearerToken=oidcTokenHelper.createTotalAccessToken(userId);
+		} else {
+			bearerToken=HttpAuthUtil.getBearerToken(req);
+			if (!isTokenEmptyOrNull(bearerToken)) {
+				try {
+					oidcTokenHelper.validateJWT(bearerToken);
+				} catch (IllegalArgumentException e) {
+					String failureReason = "Invalid access token";
+					if (StringUtils.isNotEmpty(e.getMessage())) {
+						failureReason = e.getMessage();
+					}
+					HttpAuthUtil.reject((HttpServletResponse)servletResponse, failureReason);
+					log.warn(failureReason, e);
+					return;
+				}	
+			}
 		}
-		
-		if (userId == null && !allowAnonymous) {
-			String reason = "The session token provided was missing, invalid or expired.";
-			reject(req, (HttpServletResponse) servletResponse, reason);
-			log.warn("Anonymous not allowed");
-			return;
-		}
-		
+
 		// If the user has been identified, check if they have accepted the terms of use
 		if (userId != null) {
```

Comment:
whenever we have a userId we also have a bearer token, so eliminate the case: if (userId!=null)